tests/unittests/test_datasource/test_ec2.py

```diff
@@ -358,21 +497,19 @@ def test_network_config_cached_property_refreshed_on_upgrade(self, m_dhcp):
         register_mock_metaserver(
             'http://169.254.169.254/2009-04-04/meta-data/', DEFAULT_METADATA)
         mac1 = '06:17:04:d7:26:09'  # Defined in DEFAULT_METADATA
-        get_interface_mac_path = (
-            'cloudinit.sources.DataSourceEc2.net.get_interface_mac')
+        get_interface_mac_path = M_PATH_NET + 'get_interfaces_by_mac'
         ds.fallback_nic = 'eth9'
         with mock.patch(get_interface_mac_path) as m_get_interface_mac:
-            m_get_interface_mac.return_value = mac1
+            m_get_interface_mac.return_value = {mac1: 'eth9'}
```

Comment:
This can't be right?  

def get_interface_mac(ifname):                                                 
    """Returns the string value of an interface's MAC Address""" 

it does not return a dict.

Reply:
Sorry bad variable name threw you, I'm mocking get_interfaces_by_mac in that test now 

I renamed it appropriately to 
m_get_interfaces_by_mac.return_value which does return a dict